easybuild/tools/module_naming_scheme/toolchain.py

```diff
@@ -102,23 +102,24 @@ def det_toolchain_compilers(ec):
     elif tc_elems[TOOLCHAIN_COMPILER]:
         tc_comps = []
 
-        # first consider concatenation of list of compiler module names as a single toolchain element
-        # (for example ['icc', 'ifort'] -> 'iccifort'
-        combined_comp_elem = ''.join(tc_elems[TOOLCHAIN_COMPILER])
-        combined_comp_elem_details = det_toolchain_element_details(ec.toolchain, combined_comp_elem, allow_missing=True)
-        if combined_comp_elem_details:
-            # make sure compiler details is a regular dict value
-            if isinstance(combined_comp_elem_details, EasyConfig):
-                combined_comp_elem_details = combined_comp_elem_details.asdict()
-            # add details for each compiler separately, using details from combo
-            for comp_elem in tc_elems[TOOLCHAIN_COMPILER]:
-                comp_elem_details = copy.copy(combined_comp_elem_details)
-                comp_elem_details['name'] = comp_elem
-                tc_comps.append(comp_elem_details)
-        else:
-            # consider individual compiler module names as fallback
-            for comp_elem in tc_elems[TOOLCHAIN_COMPILER]:
-                tc_comps.append(det_toolchain_element_details(ec.toolchain, comp_elem))
+        tc_comp_elems = copy.copy(tc_elems[TOOLCHAIN_COMPILER])
+
+        # for toolchains including icc & ifort (like iccifortcuda), always consider iccifort first
+        if 'icc' in tc_comp_elems and 'ifort' in tc_comp_elems:
+            combined_comp_elem_details = det_toolchain_element_details(ec.toolchain, 'iccifort', allow_missing=True)
```

Comment:
@akesandgren Strictly speaking, you're introducing a regression here by no longer considering the concatenation of toolchain compilers, we should retain that imho, and also consider `iccifort` when both `icc` and `ifort` are in the toolchain (rather than only doing the latter).

That (partially at least) comes back to the suggestion I did earlier, see https://github.com/easybuilders/easybuild-framework/pull/3055#discussion_r338175233
This could be something like:

combined_